code/LINDA/LINDA_fire.dm

```diff
@@ -151,19 +151,16 @@
 
 // Garbage collect itself by nulling reference to it
 
-/obj/effect/hotspot/proc/Kill()
+/obj/effect/hotspot/Destroy()
 	air_master.hotspots -= src
 	DestroyTurf()
-	qdel(src)
-
-/obj/effect/hotspot/Destroy()
 	set_light(0)
 	if(istype(loc, /turf/simulated))
 		var/turf/simulated/T = loc
 		if(T.active_hotspot == src)
 			T.active_hotspot = null
 	loc = null
-	return ..()
+	return QDEL_HINT_PUTINPOOL
```

Comment:
Destroy should always call its parent, even if it returns a different value in the end. And there's no need to manually set loc to null, because that will be done higher up the Destroy chain.